packages/@aws-cdk/aws-ecs/lib/container-image.ts

```diff
@@ -33,12 +44,22 @@ export abstract class ContainerImage {
    */
   public abstract readonly imageName: string;
 
+  /**
+   * Optional credentials for a private image registry
+   */
+  protected abstract readonly credentials?: RepositoryCreds;
```

Comment:
But you know what... this propery can probably be removed, no? I don't think it's used anywhere?

Reply:
(replying for context) You're right, it's a holdover from when we were checking it directlyfrom `ContainerDefinition` to make a decision re: calling `toRepositoryCredentialsJson`, but once we moved to using multiple return types this check & field became unnecessary.